python/django/security/path-traversal-file-name.yaml

```diff
@@ -0,0 +1,57 @@
+rules:
+  - id: path-traversal
+    patterns:
+      - pattern-inside: |
+          def $F(...):
+            ...
+      - pattern-not: |
+          ...
+          os.path.realpath(...)
+          ...
+      - pattern-not: |
+          ...
+          os.path.abspath(...)
+          ...
+      - pattern-either:
+        # match % use cases
+        - pattern: |
+            $V = request.$W.get($X)
+            ...
+            $FILE % ($V)
+        - pattern: |
+            $V = request.$W[$X]
+            ...
+            $FILE % ($V)
+        - pattern: |
+            $V = request.$W($X)
+            ...
+            $FILE % ($V)
+        - pattern: |
+            $V = request.$W
+            ...
+            $FILE % ($V)
+        # match format use cases
+        - pattern: |
+            $V = request.$W.get($X)
+            ...
+            $FILE.format(..., $V, ...)
+        - pattern: |
+            $V = request.$W[$X]
+            ...
+            $FILE.format(..., $V, ...)
+        - pattern: |
+            $V = request.$W($X)
+            ...
+            $FILE.format(..., $V, ...)
+        - pattern: |
+            $V = request.$W
+            ...
+            $FILE.format(..., $V, ...)
+        - pattern-where-python: |
+            vars.get('$FILE', '').split('.')[-1] in ['log', 'zip','txt', 'csv']
```

Comment:
Nice, that's really clever 👍

Reply:
Nice! Could use a set for negligible pref gains